app/src/main/res/values/dimens.xml

```diff
@@ -5,13 +5,108 @@
   <dimen name="activity_vertical_margin">16dp</dimen>
   <dimen name="dimen_small_padding">4dp</dimen>
   <dimen name="dimen_medium_padding">8dp</dimen>
+  <dimen name="dimen_big_padding">16dp</dimen>
   <dimen name="navigation_width">260dp</dimen>
   <dimen name="widget_margin">0dp</dimen>
   <dimen name="fab_margin">16dp</dimen>
   <dimen name="appbar_padding_top">8dp</dimen>
   <dimen name="abc_search_view_text_min_width">160dip</dimen>
   <dimen name="abc_dropdownitem_text_padding_left">8dip</dimen>
   <dimen name="abc_dropdownitem_text_padding_right">8dip</dimen>
+  <dimen name="ic_star_yellow_width">100dp</dimen>
+  <dimen name="ic_star_yellow_height">100dp</dimen>
+  <dimen name="activity_bookmarks_bottom_padding">15dp</dimen>
+  <dimen name="activity_bookmarks_padding_left">10dp</dimen>
+  <dimen name="activity_bookmarks_padding_right">10dp</dimen>
+  <dimen name="activity_bookmarks_padding_top">15dp</dimen>
+  <dimen name="network_permission_text_margin_top">200dp</dimen>
+  <dimen name="network_permission_button_margin_top">230dp</dimen>
+  <dimen name="library_list_padding_bottom">60dp</dimen>
+  <dimen name="bookmark_tab_icon_width">24dp</dimen>
+  <dimen name="bookmark_tab_icon_height">24dp</dimen>
+  <dimen name="bookmark_title_margin">15dp</dimen>
+  <dimen name="bookmark_title_text_size">15sp</dimen>
+  <dimen name="content_header_margin_left">16dp</dimen>
+  <dimen name="content_header_margin_right">16dp</dimen>
+  <dimen name="favicon_width">48dp</dimen>
+  <dimen name="download_item_linear_layout_width">0dp</dimen>
+  <dimen name="download_progress_padding">1dp</dimen>
+  <dimen name="pause_min_height">35dp</dimen>
+  <dimen name="pause_min_width">35dp</dimen>
+  <dimen name="stop_min_width">35dp</dimen>
+  <dimen name="stop_min_height">35dp</dimen>
+  <dimen name="stop_margin_left">8dp</dimen>
+  <dimen name="stop_margin_right">8dp</dimen>
+  <dimen name="download_management_no_downloads_padding_bottom">75dp</dimen>
+  <dimen name="download_management_no_downloads_text_size">20sp</dimen>
+  <dimen name="drawer_left_view_height">1dp</dimen>
+  <dimen name="left_drawer_list_divider_height">0dp</dimen>
+  <dimen name="left_drawer_list_height">0dp</dimen>
+  <dimen name="drawer_left_line_view_height">1dp</dimen>
+  <dimen name="action_back_button_padding_button">4dp</dimen>
+  <dimen name="action_back_button_padding_left">4dp</dimen>
+  <dimen name="action_back_button_padding_right">4dp</dimen>
+  <dimen name="action_back_button_padding_top">4dp</dimen>
+  <dimen name="plus_drawable_padding_bottom">4dp</dimen>
+  <dimen name="plus_drawable_padding_left">4dp</dimen>
+  <dimen name="plus_drawable_padding_right">4dp</dimen>
+  <dimen name="plus_drawable_padding_top">4dp</dimen>
+  <dimen name="action_forward_button_padding_bottom">4dp</dimen>
+  <dimen name="action_forward_button_padding_left">4dp</dimen>
+  <dimen name="action_forward_button_padding_right">4dp</dimen>
```

Comment:
There are so many dimensions having value ```4dp```. Can you please remove the unnecessary ones and rename the single one to a general name. Something like ```dimen_4dp``` ?
Make sure you also remove other duplications as well.

Reply:
The whole purpose of declaring different dimensions with same value for each View was to obtain the flexibility to change that respective value any time without any hassle.
@srv-twry @mhutti1 Still, shall I change those?

Reply:
@srv-twry I have made necessary changes.

Reply:
@srv-twry I think the changes you requested conflicts with the purpose of this PR. Sizes are kept in `dimes.xml` so that we can have different dimension values for different screen sizes and/or different screen orientations. Referencing `dimen_4d` instead of using the hardcoded value of `4d` only adds overhead without any substantial benefits. Read [this article](http://android4beginners.com/2013/07/appendix-c-everything-about-sizes-and-dimensions-in-android/) for more info.

Reply:
@mhutti1 please look into this. @brijeshshah13 you might have to revert commit caf35e0 based on @mhutti1's opinion.

Reply:
@RohanBh Thanks for the info and the article.

Reply:
Yeah I agree with @RohanBh lets make these changes

Reply:
@mhutti1 I agree with @RohanBh, but instead of testing all layouts and their dimensions on all devices such as phones, small tablets and big tablets, wouldn't it be better if we start using `CoordinatorLayout`. It will:

- save us our productive time

- prevent adding redundant code for every layout and dimension, resulting in saving apk size.

Your opinion?